docs/source/developers/guide/introduction.rst

```diff
@@ -0,0 +1,135 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _introduction:
+
+**********************
+New Contributor's Guide
+**********************
+
+This guide is meant to be a resource for contributing to
+`Arrow <https://arrow.apache.org//>`_ for new contributors.
+
+Starting to contribute to a project like Apache Arrow can look
+scary if not completely impossible. But we can reassure you it is
+not. What we have to do is to take small steps.
+
+
+Why contribute to Arrow?
+========================
+There can be various reasons why someone would want to contribute
+to Arrow and all of them are valid!
+
+* You find the project interesting and would like to give it a go,
+  learn and grow.
+
+* You use Arrow in the project you are working on and it would be
+  great to make some quick fixes by yourself
```

Comment:
Here as well, I would cut on the informal talk (starting from _"But we can reassure you..."_), which is nice when communicating personally with someone, but not great for written docs IMHO.